### ISSUE TITLE
VCPU 1 exceeds maximum allowed (1)

### ISSUE THREAD

**Issue by arcnmx** [64kramsystem/qemu-pinning#35]
Since updating to 6.1.0, the options that have worked previously are now failing to work with 416986f5594b6db7eb305a2e3256b70f52f1fc5f. I don't think I'm doing anything strange?

```
-smp 8,sockets=1,cores=8,threads=1
-vcpu vcpunum=0,affinity=0 -vcpu vcpunum=1,affinity=1 -vcpu vcpunum=2,affinity=2 -vcpu vcpunum=3,affinity=3 -vcpu vcpunum=4,affinity=4 -vcpu vcpunum=5,affinity=5 -vcpu vcpunum=6,affinity=6 -vcpu vcpunum=7,affinity=7

2021-10-02T20:05:09.749855Z qemu-system-x86_64: -vcpu vcpunum=1,affinity=1: VCPU 1 exceeds maximum allowed (1)
```

**Comment by 64kramsystem** [comment]
Hi! Thanks for the report (and bisection :wink:). I'll check it over this week :smiley:

**Comment by 64kramsystem** [comment]
OK. On 6.1, the initialization sequence very likely changed, so that the maximum cpus value is configured after the checks. I'm going to move the check, but I need to find out where it's the best place to do so.

**Comment by 64kramsystem** [comment]
Closed this in `master` (and `6.1.0-pinning`), by moving the check later in the boot process. It's probably not ideal (but not necessarily bad, either), but I don't have the resources to look more in depth.

**Comment by arcnmx** [comment]
I figured as much and that seems like a reasonable enough solution to me. It works again, thanks!
